City-of-Helsinki/youth-membership-admin-ui
# YM-414 | Change OIDC client to fetch multiple tokens

## Description
In this PR I am doing the initial work for transition into using the federated API. I
- configured the environment to use the federated API
- updated the types
- changed the authentication routine to use two API keys
- changed the has permission query to work with the federated API
- refactored authentication and authorisation
- refactored redirects after log in

## Context
This step allows for queries and mutations to be ported into using the new federated API.

[YM-414](https://helsinkisolutionoffice.atlassian.net/browse/YM-414)

## How Has This Been Tested?
- I ran browser tests to check that authentication works
    - I added the required permissions to the admin test user
- I added units tests for authentication and authorisation services

## Manual Testing Instructions for Reviewers
As an admin user (in `youthAdmin` group in jässäri api)

1. Try and log in
1. Expect log in to work
